sdk/src/azure/core/az_http_policy.c

```diff
@@ -48,11 +47,32 @@ AZ_NODISCARD az_result az_http_pipeline_policy_telemetry(
     az_http_request* ref_request,
     az_http_response* ref_response)
 {
+  // https://azure.github.io/azure-sdk/general_azurecore.html#telemetry-policy
+  uint8_t telemetry_id_buffer[200] = { 0 };
+  az_span telemetry_id = AZ_SPAN_FROM_BUFFER(telemetry_id_buffer);
+  {
+    _az_http_policy_telemetry_options* options = (_az_http_policy_telemetry_options*)(ref_options);
+
+    az_span remainder = az_span_copy(telemetry_id, AZ_SPAN_FROM_STR("azsdk-c"));
 
-  _az_http_policy_telemetry_options* options = (_az_http_policy_telemetry_options*)(ref_options);
+    {
+      az_span const component_name = options->component_name;
+      if (az_span_size(component_name) > 0)
```

Comment:
Add an ASSERT for when component_name is not specified.  The component_name is required

Reply:
Technically, that would be a breaking change, but currently no client has been using it.

Reply:
I'll add it.